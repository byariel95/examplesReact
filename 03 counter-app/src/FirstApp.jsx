import React from 'react'
import PropTypes from 'prop-types'
//import React, { Fragment } from 'react'


const FirstApp = ({ saludo , subtitle = 'this is the subtitle', defaultProp}) => {


   
    // const obj = {
    //     name: 'hola',
    //     other: 'zhii'
    // }

    return (
        <>
            <h1> { saludo } </h1>
            {/* <pre>{ JSON.stringify(obj,null,3)}</pre> */}
            <p>{ subtitle }</p>
            <p>{ defaultProp}</p>
        </>
        /*<Fragment>
            <h1>mi titulo </h1>
            <p>Hola Mundo desde un functional component </p>        
        </Fragment>*/    
    );
};


FirstApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    defaultProp : "hola"
}

export default FirstApp;