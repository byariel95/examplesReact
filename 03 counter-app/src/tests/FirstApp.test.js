import React from 'react';
import { render} from '@testing-library/react';
import FirstApp from '../FirstApp';
import '@testing-library/jest-dom';
import {shallow } from 'enzyme';


describe('prueba en componente <FirstAPP>', () => {
    
    /*test(' debe mostrar el msj de hola soy goku', () => {
         const saludo = 'Hola, soy Goku';
         const { getByText} = render( <FirstApp saludo={saludo}/>);

         expect( getByText( saludo )).toBeInTheDocument();
    })*/

    test('debe mostrar <>firstApp/>  ', () => {
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow( <FirstApp saludo={saludo} />);
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('debera mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola, soy Goku';
        const subTitle = 'soy un subtitulo'
        const wrapper = shallow( 
        <FirstApp 
        saludo={saludo}
        subtitle={subTitle} 
        />
        );

        const texto = wrapper.find('h2').text();
        expect(texto).toBe(subTitle);
    })
    

})
