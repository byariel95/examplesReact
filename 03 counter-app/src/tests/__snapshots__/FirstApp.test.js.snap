// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en componente <FirstAPP> debe mostrar <>firstApp/>   1`] = `
<Fragment>
  <h1>
     
    Hola, soy Goku
    !! 
  </h1>
  <h2>
    this is the subtitle
  </h2>
  <p>
    hola
  </p>
</Fragment>
`;

exports[`prueba en componente <FirstAPP> should  1`] = `
<Fragment>
  <h1>
     
    Hola, soy Goku
    !! 
  </h1>
  <h2>
    this is the subtitle
  </h2>
  <p>
    hola
  </p>
</Fragment>
`;
